### DISCUSSION TITLE
How do I make a GitHub action PR show as Verified?

### DISCUSSION THREAD

**Discussion by fmunteanu** [community/community#147754] [Actions] [answered]
### Select Topic Area

Question

### Body

I use a GitHub Action that updates the repo documentation (see PR [example](https://github.com/axivo/k3s-cluster/pull/339)):

```
      - name: Update documentation
        run: |
          git config user.email '${{ github.actor_id }}+${{ github.actor }}@users.noreply.github.com'
          git config user.name '${{ github.actor }}'
          git fetch
          git switch ${{ github.head_ref }}
          helm-docs --values-file=./defaults/main.yaml
          if [ -n "$(git status --porcelain)" ]; then
            git add -A
            git commit -m 'docs(github-action): update documentation'
            git push origin ${{ github.head_ref }}
          fi
```

As you can see, Renovate's PR shows as verified:

![image](https://github.com/user-attachments/assets/d0edb22c-9179-4a09-895c-21055745760a)

I would like to have the same result for `docs(github-action)` PR, so I can enable the `Verified` status as requirement into branch protection.

**Comment by mecodeatlas** [comment]
Thanks for posting in the GitHub Community, @fmunteanu!

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Discussions category is solely related to conversations around the [GitHub product Discussions](https://docs.github.com/en/discussions/collaborating-with-your-community-using-discussions/about-discussions). This question should be in the Actions category. I've gone ahead and moved it for you. Good luck!

**Comment by fmunteanu** [comment, marked as answer]
I ended-up using the GitHub GraphQL API with `actions/github-script` into workflow, instead of actually using the binary. The step handles properly both additions and deletions, with `CreateCommitOnBranch` mutation.

Previous implementation, without verified signature:

![H3BqjVaO](https://github.com/user-attachments/assets/a8a9348c-4720-4d37-8ab1-993897ba7709)

New implementation:

![UPC7phED](https://github.com/user-attachments/assets/40366379-163a-4b40-b58c-18ac947f3f25)

You can examine the [actual PR](https://github.com/axivo/k3s-cluster/pull/349) with the new implementation.
